### PR TITLE
Add gulp eslint task

### PR DESCRIPTION
**Pull request by marcusmolchany**
This change is for issue #134

This change adds the gulp task for eslint. Currently it is just
using the default `eslint()` configuration, but the configuration
is customizable in the gulp task as well as adding .eslintrc files
at any level of the directory structure.

Curently there are 12 errors after running eslint. Not sure why these
are errors as the syntax is correct.

Once this test is working completely it can be added to the default
task set or continuous integration task set.

Usage:
  `$ gulp eslint`

Documentation:
  https://github.com/adametry/gulp-eslint
  http://eslint.org/docs/user-guide/configuring

### CODE REVIEW

**Comment by marcusmolchany** [comment]
let me know if you'd rather have these squashed into one commit

from the last commit message, there are errors when I ran the gulp task but the syntax in every file looked correct so not sure why it was failing. Tried to make a top-levl `.eslintrc` file like this
```json
{
  "env": {
    "es6": true
  }
}
```

but that didn't seem to help

File: gulp/tasks/eslint.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import config from '../config';
+import gulp   from 'gulp';
+import eslint from 'gulp-eslint';
+
+gulp.task('eslint', () => {
+  // ESLint ignores files with "node_modules" paths.
+  // So, it's best to have gulp ignore the directory as well.
+  // Also, Be sure to return the stream from the task;
+  // Otherwise, the task may end before the stream has finished.
+  return gulp.src([config.scripts.src, '!app/js/templates.js', '!node_modules/**'])
```
Comment (jakemmarsh) [inline gulp/tasks/eslint.js]: You shouldn't need to explicitly ignore `node_modules` since you're only sourcing `config.scripts.src`

**Comment by jakemmarsh** [comment]
Your CI failure was unrelated to your PR, and the issue should be fixed in master. You mind merging master into your branch?

Minus one small note I just left, the PR itself looks good. Thanks!

File: gulp/tasks/eslint.js (hunk 1 above)
Comment (marcusmolchany) [inline gulp/tasks/eslint.js]: done

**Comment by marcusmolchany** [comment]
hm, i rebased this on top of origin/master not sure whats going on with ci

**Comment by jakemmarsh** [comment]
The issue with CI is it's attempting to run the protractor tests via SauceLabs on all PRs, which isn't possible via Travis. I've just put out another possible fix in 5ebc94b3885b4f7ad78907fa275568aced3ed714. Unfortunately it's difficult for me to test because the issue only affects external PRs.

If you wouldn't mind merging master into your branch once more, hopefully the issue is fixed!

**Comment by marcusmolchany** [comment]
yea for sure gimmie a few minutes

**Comment by marcusmolchany** [comment]
looks like it failed again, this is one pesky bug

**Comment by jakemmarsh** [comment]
Hah yup, very annoying. Your PR looks good though so I'm just going to go ahead and merge it. Thanks!

**Pull request merged**